app/frontend/src/queryKeys.ts

```diff
@@ -79,6 +81,32 @@ export const communityEventsKey = (communityId: number) => [
   communityId,
 ];
 
+export const eventKey = (eventId: number) => ["event", eventId];
+export interface EventUsersInput {
+  eventId: number;
+  type: QueryType;
+}
+export function isEventUsersInputType(
+  queryKey: unknown
+): queryKey is EventUsersInput {
+  return (
+    typeof queryKey === "object" &&
+    queryKey !== null &&
+    queryKey.hasOwnProperty("eventId") &&
+    queryKey.hasOwnProperty("type")
+  );
+}
```

Comment:
not a fan :P

Reply:
It's the only way to do it type safe but I probably won't need this function after going with your suggestion